packages/noya-state/src/selectors/pointSelectors.ts

```diff
@@ -57,3 +57,49 @@ export const getSelectedPoints = (
 
   return points;
 };
+
+export const getSelectedControlPoint = (
```

Comment:
It looks like this is about the same as `getSelectedPoints`. Can we try sharing the main logic?

For example, a new function that they both call, 

```ts
const getSelectedPointsFromPointLists = (
  state: ApplicationState,
  pointLists: Record<string, number[]>,
): DecodedCurvePoint[] => {
  // ...
};
```

When we have a `selectedControlPoint`, we can make a new "pointsList" with just one point, `{ [layerId]: [pointIndex] }`, and then take just the first result from that function. Which is pretty similar to what you have now.